bitnami/odoo/templates/deployment.yaml

```
@@ -88,6 +88,9 @@ spec:
           volumeMounts:
             - name: odoo-data
               mountPath: /bitnami/odoo
+              {{- if .Values.persistence.useSubPath | default false }}
+              subPath: odoo
+              {{- end }}
```

Instead of hardcoding the `odoo` folder we could apply this approach:
```suggestion
              {{- if .Values.persistence.subPath }}
              subPath: {{ .Values.persistence.subPath }}
              {{- end }}
```
As we have defined in our template: https://github.com/bitnami/charts/blob/e1e00f59dce468e40b0d8d7b1d88fe8630dea6a7/template/CHART_NAME/templates/deployment.yaml#L85-L90

---

Hi, thanks for the feedback.
That was my first approach as well, however, there are other places where the path is hardcoded.
Not only on the chart, but within the container image scripts.
Making it configurable might silently break other components, making it harder to debug.

Still, if the maintainers prefer that, I can do it in no time.
What do you prefer?

---

@fmulero could you please provide guidance about how to proceed?
Thanks!

---

Sorry about the delay @baldo-io.

> That was my first approach as well, however, there are other places where the path is hardcoded.
> Not only on the chart, but within the container image scripts.
> Making it configurable might silently break other components, making it harder to debug.

The solution proposed should work. `subPath` affects to the volume, not the `mountPath`, so the container shouldn't  know about the name of the folder in the volume.

Please give it a try and share any though or finding.

---

No worries @fmulero , thanks for the response.
I am running the proposed solution in OCI/OKE production already.
It solves the initial bug reported.

---

Sorry I didn't explain it well. I mean the solution I proposed, the one existing in our templates.

The `subPath` folder is relative to the volume, so we can call it with the name we want. The container won't know about that name because will be mounted in the `mountPath` especified. For that reason we prefer to use the value `persistence.subPath`, to give more freedom to the users